Add validation for max APP number in AssignNumberModal

ChinemeremChigbo opened the pull request:
## Implementation description
* Introduced a MAX_APP_NUMBER constant and updated validation logic to ensure the assigned number is an integer and does not exceed 2,147,483,647.
<img width="495" height="341" alt="image" src="https://github.com/user-attachments/assets/d0b4ddba-0f2e-4de4-a4b1-49ae62c4febc" />
<img width="427" height="251" alt="image" src="https://github.com/user-attachments/assets/16bc59e9-4fe7-404e-a375-eded469e47f7" />

ChinemeremChigbo committed:
Add validation for max APP number in AssignNumberModal

Introduced a MAX_APP_NUMBER constant and updated validation logic to ensure the assigned number is an integer and does not exceed 2,147,483,647.

ssreekar reviewed (approved):
LGTM, one note to add is that it would be nice to have a min_APP_NUMBER as well, for -1 cases. Approving regardless as that might be out of scope for this ticket, leaving up to your disgression

ChinemeremChigbo committed:
Add minimum value validation for APP number input

Introduces a MIN_APP_NUMBER constant and enforces that the assigned APP number cannot be less than zero. Updates the NumberInput component to include min and max attributes for better user input validation.
